File: crates/cxx-qt-lib-extras/src/core/qlocale.cpp
// clang-format off
// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
// clang-format on
// SPDX-FileContributor: Nicolas Fella <nicolas.fella@kdab.com>
//
// SPDX-License-Identifier: MIT OR Apache-2.0
#include "cxx-qt-lib-extras/qlocale.h"
#include <cxx-qt-lib/assertion_utils.h>

assert_alignment_and_size(QLocale, { ::std::size_t a0; });

static_assert(!::std::is_trivially_copy_assignable<QLocale>::value);
static_assert(!::std::is_trivially_copy_constructible<QLocale>::value);

static_assert(!::std::is_trivially_destructible<QLocale>::value);

static_assert(QTypeInfo<QLocale>::isRelocatable);

File: tests/qt_types_standalone/rust/src/qlocale.rs
// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
// SPDX-FileContributor: Nicolas Fella <nicolas.fella@kdab.com>
//
// SPDX-License-Identifier: MIT OR Apache-2.0

use cxx_qt_lib_extras::QLocale;

#[cxx::bridge]
mod qlocale_cxx {
    unsafe extern "C++" {
        include!("cxx-qt-lib-extras/qlocale.h");
        type QLocale = cxx_qt_lib_extras::QLocale;
    }

    extern "Rust" {
        fn construct_qlocale() -> QLocale;
        fn clone_qlocale(l: &QLocale) -> QLocale;
    }
}

fn construct_qlocale() -> QLocale {
    QLocale::default()
}

fn clone_qlocale(l: &QLocale) -> QLocale {
    l.clone()
}
